src/Yaapii.Atoms/IO/TempDirectory.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.IO;
+
+namespace Yaapii.Atoms.IO
+{
+    public sealed class TempDirectory : IScalar<DirectoryInfo>, IDisposable
+    {
+        private readonly Uri path;
```

Comment:
do not use URI for a directory. URI is Uniform Resource Identifier. It does not point to directories.